src/main/java/org/opensearch/security/support/Base64CustomHelper.java

```diff
@@ -0,0 +1,220 @@
+
+        Preconditions.checkArgument(object != null, "object must not be null");
+        final BytesStreamOutput streamOutput = new SafeBytesStreamOutput(128);
+        Class<?> clazz = object.getClass();
+        try {
+            prohibitUnsafeClasses(clazz);
+            CustomSerializationFormat customSerializationFormat = getCustomSerializationMode(clazz);
+            switch (customSerializationFormat) {
+                case WRITEABLE:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.WRITEABLE.id);
+                    streamOutput.writeByte((byte) getWriteableClassID(clazz).intValue());
+                    ((Writeable) object).writeTo(streamOutput);
+                    break;
+                case STREAMABLE:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.STREAMABLE.id);
+                    streamableRegistry.writeTo(streamOutput, object);
+                    break;
+                case GENERIC:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.GENERIC.id);
+                    streamOutput.writeGenericValue(object);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                        String.format("Could not determine custom serialization mode for class %s", clazz.getName())
+                    );
+            }
+        } catch (final Exception e) {
+            throw new OpenSearchException("Instance {} of class {} is not serializable", e, object, object.getClass());
+        }
+        final byte[] bytes = streamOutput.bytes().toBytesRef().bytes;
+        streamOutput.close();
+        return BaseEncoding.base64().encode(bytes);
+    }
+
+    protected static Serializable deserializeObject(final String string) {
+
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(string), "string must not be null or empty");
+        final byte[] bytes = BaseEncoding.base64().decode(string);
+        try (final BytesStreamInput streamInput = new SafeBytesStreamInput(bytes)) {
+            CustomSerializationFormat serializationFormat = CustomSerializationFormat.fromId(streamInput.readByte());
+            switch (serializationFormat) {
+                case WRITEABLE:
+                    final int classId = streamInput.readByte();
+                    Class<?> clazz = getWriteableClassFromId(classId);
+                    prohibitUnsafeClasses(clazz);
+                    return (Serializable) clazz.getConstructor(StreamInput.class).newInstance(streamInput);
+                case STREAMABLE:
+                    return (Serializable) streamableRegistry.readFrom(streamInput);
+                case GENERIC:
+                    return (Serializable) streamInput.readGenericValue();
+                default:
+                    throw new IllegalArgumentException("Could not determine custom deserialization mode");
+            }
```

Comment:
```suggestion
            Serializable obj = null;
            switch (serializationFormat) {
                case WRITEABLE:
                    final int classId = streamInput.readByte();
                    Class<?> clazz = getWriteableClassFromId(classId);
                    obj = (Serializable) clazz.getConstructor(StreamInput.class).newInstance(streamInput);
                case STREAMABLE:
                    obj = (Serializable) streamableRegistry.readFrom(streamInput);
                case GENERIC:
                    obj = (Serializable) streamInput.readGenericValue();
                default:
                    throw new IllegalArgumentException("Could not determine custom deserialization mode");
            }
            prohibitUnsafeClasses(obj.getClass());
            return obj;
```

Reply:
Resolving. This suggestion has been incorporated.